components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1093,7 +1093,11 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         const context: LogContext = { userId: user.id, workspaceId };
         log.info(context, 'watchHeadlessWorkspaceLogs', { workspaceId });
 
-        const { instance, workspace } = await this.internGetCurrentWorkspaceInstance(user.id, workspaceId);
+        const { instance, workspace } = await this.internGetCurrentWorkspaceInstance(user.id, workspaceId)
+        if (this.disposables.disposed) {
```

Comment:
does this mean that the check for `disposed` is potentially missing in other places? relying on the `dispose` from disposablecollection instead of an additional flag could lead easier to bugs again, just because it's flipped by a following `disposables.push`, right?

Reply:
> does this mean that the check for disposed is potentially missing in other places?

We need to check in methods which awaits before pushing to the collection. If the request method does not use awaits then pushing to the collection directly is fine. I could not find any other such places at least in core.

>  just because it's flipped by a following disposables.push, right?

In this case the collection always have at least one element before `dispose` is called. We add this element as first thing in the initialise method.

Reply:
anyway I am going to replace it with `if (this.client)`

Reply:
ah, found another place watchWorkspaceImageBuildLogs